client/router.js

```diff
@@ -369,3 +427,56 @@ Router.prototype._updateCallbacks = function () {
 
 Router.prototype._page = page;
 Router.prototype._qs = qs;
+
+function shouldCallTrigger(current, fn) {
+  var name = current.route.name;
+
+  if (typeof fn !== 'function') {
+    return false;
+  }
+
+  if (!fn._only && !name) {
```

Comment:
How about if we branch these multiples ifs into three.

~~~js
if(fn._only) {

} else if(fn._except) {

} else {

}
~~~